### PR TITLE
Api

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
API default.py file for webservices

### CODE REVIEW

**Comment by chemistry-sourabh** [comment]
@ravisantoshgudimetla @apoorvemohan  updated PR

File: docs/rest_api.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+ "img" : "<image_name>" ,
+ "network" : "<network_name>" ,
+ "channel" : "<channel in haas>" ,
+ "nic" : "<nic to connect on>"
+}
+```
+
+####Responses:
+* 200. This means the provision call is successful.
+* Internal 500 with some junk characters. This means the request body is not proper.
```
Comment (knikolla) [inline docs/rest_api.md]: 500 means Internal Server Error. It means the request was proper but the server failed to process it for some fault of the server, not the client or the request.

The correct return code here is [400](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error).

File: docs/rest_api.md [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+* 401. This means unauthorized access to ceph image or snapshot or image already exists in ceph.
+* 403. This means a ceph connection problem.
+* 409. Image busy exception.
+* 444. You used a wrong request method like PUT instead of POST etc.
```
Comment (knikolla) [inline docs/rest_api.md]: [405 - Method not allowed](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_Client_Error).

File: docs/rest_api.md [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+###Provision:
+Provision API is needed for provisioning a node from MOC cluster as of now. Provision operation internally calls a ceph clone operation. Ceph clone operation usually takes a snap_
+
+Following is the call for API:
+
+####Link:
+http://BMI_SERVER:PORT/provision/
+
+####Request Type:
+PUT
```
Comment (knikolla) [inline docs/rest_api.md]: Technically, the correct method in these cases should have been POST. See [here](http://restcookbook.com/HTTP%20Methods/put-vs-post/).

File: docs/rest_api.md [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+ "network" : "<network_name>" ,
+ "channel" : "<channel in haas>" ,
+ "nic" : "<nic to connect on>"
+}
+```
+
+####Responses:
+* 200. This means the provision call is successful.
+* Internal 500 with some junk characters. This means the request body is not proper.
+* 401. This means unauthorized access to ceph image or snapshot or image already exists in ceph.
```
Comment (knikolla) [inline docs/rest_api.md]: This should be 403 Forbidden.

File: docs/rest_api.md [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+* 403. This means a ceph connection problem.
```
Comment (knikolla) [inline docs/rest_api.md]: This is clearly a server error, so 500.

**Comment by naved001** [comment]
I am closing this because we merged Sourabh's TGT branch to the dev branch #20

**Pull request closed without merging**